docs/admin/cluster-components.md

```diff
@@ -0,0 +1,110 @@
+tree only. If you are using a released version of Kubernetes, you almost
+certainly want the docs that go with that version.</h1>
+
+<strong>Documentation for specific releases can be found at
+[releases.k8s.io](http://releases.k8s.io).</strong>
+
+![WARNING](http://kubernetes.io/img/warning.png)
+![WARNING](http://kubernetes.io/img/warning.png)
+![WARNING](http://kubernetes.io/img/warning.png)
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+# Kubernetes Cluster Admin Guide: Cluster Components
+
+This document outlines the various binary components that need to run to
+deliver a functioning Kubernetes cluster.
+
+## Master Components
+
+Master components are those that provide the cluster's control plane, make
+global decisions (i.e., scheduling) about the cluster, and detect and respond to
+cluster events (e.g., starting up a new pod when a replication controller's
+'replicas' field is unsatisfied).
+
+Master components could in theory be run on any node in the cluster. However,
+for simplicity, current set up scripts typically start all master components on
+the same VM. (But see [high-availability](availability.md) for an example
+multi-master-VM setup.)
+
+Even in the future, when Kubernetes is fully self-hosting, it will probably be
+wise to only allow master components to schedule on a subset of nodes, to limit
+co-running with user-run pods, reducing the possible scope of a
+node-compromising security exploit.
+
+### kube-apiserver
+
+```kube-apiserver``` is the global point of contact for the api. It is designed to
+scale horizontally (i.e., one scales it by running more of them).
+
+### etcd
+
+```etcd``` is used as Kubernetes' backing store. All cluster data is stored here.
+Proper administration of a Kubernetes cluster includes a backup plan for etcd's
+data.
+
+### kube-controller-manager
+
+```kube-controller-manager``` is a binary that runs controllers, or the background
+processes that handle routine tasks in the cluster.
+
+These controllers include:
+
+* Node Controller
+ * Responsible for noticing & responding when nodes go down.
+* Replication Controller
+ * Responsible for maintaining the correct number of pods for every replication
+   controller in the system.
+* Endpoints Controller
+ * Populates the Endpoints object (i.e., join Services & Pods).
+* Service Account & Token Controllers
+ * Create default accounts and API access tokens for new namespaces.
+* ... and others.
+
+### kube-scheduler
+
+```kube-scheduler``` watches newly created pods that have no node, and selects a node
+for them to run on.
+
+## Node components
+
+Node components run on every node, maintaining running pods and providing them
+the Kubernetes running environment.
+
+### kubelet
+
+```kubelet``` is the primary node agent. It watches for pods that have been
+assigned to its node and runs them.
+
+### kube-proxy
+
+```kube-proxy``` is responsible for maintaining network rules, enabling the
```

Comment:
I'd reword this slightly, maybe something like "enables the Kubernetes service abstraction by maintaining network rules on the host and performing connection forwarding"